src/js/edu-benefits/definitions/veteranId.js

```diff
@@ -0,0 +1,44 @@
+import _ from 'lodash/fp';
+
+import * as ssn from '../../common/schemaform/definitions/ssn';
+
+// TODO: move vaFileNumber out of 1995 and into a shared backend definition
+import fullSchema1995 from 'vets-json-schema/dist/change-of-program-schema.json';
+const {
+  vaFileNumber
+} = fullSchema1995.definitions;
+
+
+export const uiSchema = {
+  veteranSocialSecurityNumber: _.assign(ssn.uiSchema, {
+    'ui:required': (form) => !form.veteranId['view:noSSN']
```

Comment:
@jbalboni I'm tracking down why this change breaks a bunch of unit tests, and I think it's this line. Seems to work fine in the app (I inspected the code [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/common/schemaform/reducers/index.js#L36) and [here](https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/js/common/schemaform/helpers.js#L329) and all looks fine)

I'm wondering why `form.veteranId` appears to be undefined in [DefinitionTester](https://github.com/department-of-veterans-affairs/vets-website/blob/master/test/util/schemaform-utils.jsx#L22) and if you have any ideas for what to investigate.

Also curious how you debug test failures, other than intuition :)

Here's the stack trace for one of the failing tests:
```
     TypeError: Cannot read property 'view:noSSN' of undefined
      at Object.uiRequired [as ui:required] (src/js/edu-benefits/definitions/veteranId.js:14:31)
      at src/js/common/schemaform/helpers.js:329:28
      at Array.reduce (native)
      at updateRequiredFields (src/js/common/schemaform/helpers.js:326:56)
      at src/js/common/schemaform/helpers.js:346:28
      at Array.reduce (native)
      at updateRequiredFields (src/js/common/schemaform/helpers.js:344:54)
      at new DefinitionTester (test/util/schemaform-utils.jsx:22:18)
```